core/src/main/java/org/apache/druid/java/util/common/parsers/JSONPathSpec.java

```diff
@@ -28,19 +28,23 @@
 
 public class JSONPathSpec
 {
-  public static final JSONPathSpec DEFAULT = new JSONPathSpec(null, null);
+  public static final JSONPathSpec DEFAULT = new JSONPathSpec(null, null, null);
 
   private final boolean useFieldDiscovery;
   private final List<JSONPathFieldSpec> fields;
+  private final boolean keepNullColumns;
 
   @JsonCreator
   public JSONPathSpec(
       @JsonProperty("useFieldDiscovery") Boolean useFieldDiscovery,
-      @JsonProperty("fields") List<JSONPathFieldSpec> fields
+      @JsonProperty("fields") List<JSONPathFieldSpec> fields,
+      @JsonProperty("keepNullColumns") Boolean keepNullColumns
```

Comment:
This property is used only in `JSONFlattenMaker` while this class is used by all nested formats. It would be better to have it in `JsonInputFormat` and `JSONParseSpec` instead.